web3/contract/utils.py

```diff
@@ -129,6 +132,10 @@ def call_contract_function(
     )
     normalized_data = map_abi_data(_normalizers, output_types, output_data)
 
+    if decode_tuples:
+        decoded = named_tree(fn_abi["outputs"], normalized_data)
+        normalized_data = dict_to_namedtuple(decoded)
```

Comment:
nit: since we're using these three lines in two different places, I'd slightly prefer to pull them out into their own function. But feel free to take or leave that too. I know some people consider 3 uses the magic number :)

Reply:
I think that because it's once for sync and once for async, I'll leave it as is this time. I can't imagine where it might split again, but :shrug: